### PR TITLE
[components]: utilities: ymodem: Add ymodem send protocol and ry_sy application.

### PR DESCRIPTION
**Pull request by Steven-LiuSF**
## 拉取/合并请求描述：(PR description)

[
Achieve the function of sending and receiving files via UART using YMODEM。

Use the command "ry [uart0]" to receive the file. Use the command "sy file_path [uart0]" to send the file. The console is used by default. You can choose any uart if you want.
The test uses the console to send and receive files including text, PDF, mp3 without errors, compared with the original file. (The 0x1A at the end of the file is added by the Ymodem protocol to simulate the ctrl Z, which is normal.) Other uart also passed the test. The test uses a baud rate of 115200 and stable transmission rate at 4kB / s. It has been found that an error will occur when the baud rate reaches 1.5M.

Optimization direction:
(1) Add 1024 Byte data packet support.
(2) Support higher serial baud rate transmission.
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

File: components/utilities/Kconfig [hunk 1]
```diff
@@ -2,8 +2,15 @@ menu "Utilities"
 
 config RT_USING_RYM
     bool "Enable Ymodem"
+    depends on RT_USING_DFS = y
```
Comment (armink) [inline components/utilities/Kconfig]: 收发文件是不是单独做一个宏来控制比较合适呢？

以前的 Ymodem 常用于 OTA ，对于 DFS 没有依赖，这样改以后，OTA 使用就会有问题了。

Comment (Steven-LiuSF) [reply components/utilities/Kconfig]: 这个您可以按照您的想法进行修改。我们主要用收发文件的功能，没有用到OTA，所以就没做这方面的考量。

Comment (Steven-LiuSF) [reply components/utilities/Kconfig]: 这一处是上一版提交中根据BernardXiong的建议“It's related with file system, please add RT_USING_DFS for dependent.”进行修改的。

Comment (BernardXiong) [reply components/utilities/Kconfig]: 这部分我来修改吧，感觉是没理解我的意思

Comment (Steven-LiuSF) [reply components/utilities/Kconfig]: 好的，麻烦您了

**Pull request merged**